src/ldp/AuthenticatedLdpHandler.ts

```diff
@@ -0,0 +1,145 @@
+   */
+  public constructor (args: AuthenticatedLdpHandlerArgs) {
+    Object.assign(this, args);
+  }
+
+  /**
+   * Checks if the incoming request can be handled. The check is very non-restrictive and will usually be true.
+   * It is based on whether the incoming request can be parsed to an operation.
+   * @param input - Incoming request and response. Only the request will be used.
+   *
+   * @returns A promise resolving if this request can be handled, otherwise rejecting with an Error.
+   */
+  public async canHandle (input: { request: HttpRequest; response: HttpResponse }): Promise<void> {
+    return this.requestParser.canHandle(input.request);
+  }
+
+  /**
+   * Handles the incoming request and writes out the response.
+   * This includes the following steps:
+   *  - Parsing the request to an Operation.
+   *  - Extracting credentials from the request.
+   *  - Extracting the required permissions.
+   *  - Validating if this operation is allowed.
+   *  - Executing the operation.
+   *  - Writing out the response.
+   * @param input - The incoming request and response object to write to.
+   *
+   * @returns A promise resolving when the handling is finished.
+   */
+  public async handle (input: { request: HttpRequest; response: HttpResponse }): Promise<void> {
+    let err: Error = null;
+    let operation: Operation = null;
+
+    try {
+      operation = await this.runHandlers(input.request);
+    } catch (error) {
+      err = error;
+    }
+
+    const writeData = { response: input.response, operation, error: err };
+
+    await this.responseWriter.canHandle(writeData);
+
+    return this.responseWriter.handle(writeData);
+  }
+
+  /**
+   * Runs all handlers except writing the output to the response.
+   * This because any errors thrown here have an impact on the response.
+   * @param request - Incoming request.
+   *
+   * @returns A promise resolving to the generated Operation.
+   */
+  private async runHandlers (request: HttpRequest): Promise<Operation> {
+    const op: Operation = await this.runHandler(this.requestParser, request);
+
+    const credentials: Credentials = await this.runHandler(this.credentialsExtractor, request);
+
+    const permissions: PermissionSet = await this.runHandler(this.permissionsExtractor, op);
+
+    await this.runHandler(this.authorizer, { credentials, identifier: op.target, permissions });
+
+    await this.runHandler(this.operationHandler, op);
+
+    return op;
+  }
+
+  /**
+   * Checks if a handler can handle a certain input and then lets the handler handle it handily.
+   * If not, an Error will be thrown. Specifically the error of the handler's canHandle request.
+   * @param handler - Handler to run.
+   * @param data - Data to handle.
+   *
+   * @returns The result of the handle function of the handler.
+   */
+  // eslint-disable-next-line class-methods-use-this
+  private async runHandler<T extends AsyncHandler<TIn, TOut>, TIn, TOut>(handler: T, data: TIn): Promise<TOut> {
```

Comment:
Ah, but this is this simplification I talked about in a previous comment :-)

I would follow the eslint rule here, and move this to a static helper method. You could place this in `AsyncHandler`.

Reply:
Ah yea, moving it to AsyncHandler as static would make sense. Just didn't like it as a static function for this specific class.